keyring.go

```diff
@@ -372,17 +373,28 @@ func (d *DirectKeyFetcher) FetchKeys(
 	}
 
 	results := map[PublicKeyLookupRequest]PublicKeyLookupResult{}
-	for server := range byServer {
-		// TODO: make these requests in parallel
-		serverResults, err := d.fetchKeysForServer(ctx, server)
-		if err != nil {
-			// TODO: Should we actually be erroring here? or should we just drop those keys from the result map?
-			return nil, err
-		}
-		for req, keys := range serverResults {
-			results[req] = keys
-		}
+	var resultsMutex sync.Mutex
+	var wait sync.WaitGroup
+	wait.Add(len(byServer))
```

Comment:
If we have many servers this will be a problem. Spawning hundreds of parallel HTTP requests might hit fdlimits or cause other nasty behaviour which we won't see until we hit that threshold. A worker pool would be a better solution.

Reply:
That doesn't have to happen now, but a TODO would be good.